kitsune/search/v2/base.py

```diff
@@ -22,18 +22,18 @@ def prepare(cls, instance):
 
             # This will allow child classes to have their own methods in the form of prepare_field
             prepare_method = getattr(obj, "prepare_{}".format(f), None)
-            # Ovewrite the value if there is a specific method implented for the specific field
+            # Overwrite the value if there is a specific method implemented for the specific field
             if prepare_method is not None:
                 value = prepare_method(instance)
             else:
-                value = getattr(instance, f)
+                value = cls.get_field_value(instance, f)
 
-            # Assign values to each field. Assign a dictionary to Objects
-            if isinstance(doc_mapping.resolve_field(f), field.Object):
+            # Assign values to each field. Assign a dictionary to multi locale Objects
+            field_type = doc_mapping.resolve_field(f)
+            if isinstance(field_type, field.Object) and "en-US" in field_type._mapping:
```

Comment:
Why do we need the `en-US` check?

Reply:
To distinguish a multi-locale object (which will have a "en-US" parameter) from a "normal" object, like a UserInnerDoc

Reply:
This feels a bit hacky.  I would suggest to at least use `locale` here for the check.

```python
 if isinstance(field_type, field.Object) and "en-US" in field_type._mapping:
    if locale := obj.prepare_locale(instance) and locale in field_type._mapping:
       if value:
           obj[f] = {locale:value}
    else:
       other_type_of_objects
```

Reply:
I turned the problem on its head and came up with a less hacky solution - we know we're dealing with a locale field if the ES type is Object but the prepared type is something like a String (that is to say, not an InnerDoc).